core/station/impl/src/services/external_canister.rs

```diff
@@ -517,196 +520,246 @@ impl ExternalCanisterService {
+                            AddRequestPolicyOperationInput {
+                                rule: updated_change_policy.rule,
+                                specifier: RequestSpecifier::ChangeExternalCanister(
+                                    ExternalCanisterId::Canister(external_canister.canister_id),
+                                ),
+                            },
+                        )?;
+                    }
+                }
+            }
         }
 
-        // add or update the `Change` policies
-        for policy in input.change {
-            match policy.policy_id {
-                Some(policy_id) => {
-                    if !current_policies.contains(&policy_id) {
-                        print(format!(
-                            "Policy with id {} not found for external canister {}",
-                            Uuid::from_bytes(policy_id).hyphenated(),
-                            external_canister.canister_id.to_text()
-                        ));
+        // manage the policies for the `Call` requests
+        if let Some(updated_call_policies) = input.calls {
+            match &updated_call_policies {
+                ExternalCanisterChangeCallRequestPoliciesInput::ReplaceAllBy(calls) => self
+                    .maybe_mutate_canister_calls_request_policies(
+                        external_canister,
+                        &current_policies.calls,
+                        calls,
+                    )?,
+                ExternalCanisterChangeCallRequestPoliciesInput::OverrideSpecifiedByExecutionMethods(
+                    calls,
+                ) => {
+                    // first aggregates the calls by the execution method
+                    let calls_by_execution_method = calls.iter().fold(
+                        HashMap::new(),
+                        |mut acc: HashMap<
+                            String,
+                            Vec<ExternalCanisterCallRequestPolicyRuleInput>,
+                        >,
+                         call| {
+                            acc.entry(call.execution_method.clone())
+                                .or_default()
+                                .push(call.clone());
+                            acc
+                        },
+                    );
 
-                        continue;
+                    for (method, calls) in calls_by_execution_method {
+                        self.maybe_mutate_canister_calls_request_policies(
+                            external_canister,
+                            &self
+                                .request_policy_repository
+                                .find_external_canister_call_policies_by_execution_method(
+                                    &external_canister.canister_id,
+                                    &method,
+                                ),
+                            &calls,
+                        )?;
                     }
```

Comment:
same here: we should add this back and ideally fail the request instead of continuing